client/db/bolt/db.go

```diff
@@ -1624,6 +1624,167 @@ func (idx *timeIndexNewest) add(t uint64, k []byte, b *bbolt.Bucket) {
 	})
 }
 
+// DeleteInactiveOrders deletes orders that no longer needed for normal
+// operations. Optionally accepts a time to delete orders with a later time
+// stamp. Accepts an optional function to perform on batches of deleted orders.
+func (db *BoltDB) DeleteInactiveOrders(ctx context.Context, olderThan *time.Time, perBatchFn func(ords []*db.MetaOrder) error) error {
+	start := time.Now()
+	const batchSize = 1000
+	var (
+		finished   bool
+		olderThanB []byte
+	)
+	if olderThan != nil {
+		olderThanB = uint64Bytes(encode.UnixMilliU(*olderThan))
+	} else {
+		olderThanB = uint64Bytes(encode.UnixMilliU(start))
+	}
+	for !finished {
+		if err := ctx.Err(); err != nil {
+			return err
+		}
+		deletedOrders := make([]*dexdb.MetaOrder, 0, batchSize)
+		err := db.Update(func(tx *bbolt.Tx) error {
+			// Some archived orders may still be needed for active matches.
+			// Put those order id's in a map to prevent deletion.
+			amb := tx.Bucket(activeMatchesBucket)
+			activeMatches := make(map[order.OrderID]struct{})
```

Comment:
This `activeMatches` map will generated identically on each call to `db.Update`.  Should there be a preceding `db.View`, outside of the `finished` loop, to create this map once?

Reply:
Creating outside now.